longest_palindromic_substring/longest_palindromic_substring.py

```diff
@@ -0,0 +1,48 @@
+def longest_palindromic_substring(test):
+    """
+    Function to find longest substring which is a palindrome
+
+    :param test: Input string whose substrings are to be checked
+    :return: Longest substring of input string which is a palindrome
+
+    Time complexity: O(n^2)
+    Space complexity: O(n^2)
+    n is size of input string
+    """
+    n = len(test)
+    substrings = [[False] * n] * n
+    len_palindrome = 1
+    palindrome_begin = 0
+
+    # Substrings of length 1
+    for i in range(n):
+        substrings[i][i] = True
+    # Substrings of length 2
+    for i in range(n - 1):
+        if test[i] is test[i + 1]:
+            substrings[i][i + 1] = True
+            palindrome_begin = i
+            len_palindrome = 2
+    # Substrings of length > 3
+    for i in range(3, n + 1):
+        for j in range(n - i + 1):
+            tmp = i + j - 1
+            if substrings[j + 1][tmp - 1] and test[j] is test[tmp]:
+                substrings[j][tmp] = True
+                if i > len_palindrome:
+                    len_palindrome = i
+                    palindrome_begin = j
+    return test[palindrome_begin: palindrome_begin + len_palindrome]
```

Comment:
@aashutoshrathi I've updated it with a better algorithm which takes O(1) space. Please review it.